### ISSUE TITLE
Add smart variable entity

### ISSUE THREAD

**Issue by oshtaier** [SatelliteQE/airgun#25]
Configure -> Smart Variables
* [x] All entities page
![image](https://user-images.githubusercontent.com/10895065/42396005-a8228d2e-8168-11e8-804b-0e60df1e356d.png)
* [x] Create/Update view from default page
![image](https://user-images.githubusercontent.com/10895065/42395988-96da3422-8168-11e8-9036-655b4ff4a324.png)
* [x] Create/Update view from Puppet Classes page 
![image](https://user-images.githubusercontent.com/10895065/42395963-6ecbb712-8168-11e8-8b1a-a8e8474f9a23.png)

Important notes:
- At least one valid Puppet Class should be present in the system prior any possible operation with Smart Variables permitted
- Generic detail page should be divided into sub-views as there are many test cases that like to operate only with matchers for example

**Comment by oshtaier** [comment]
Ok, that will be more challenging that I thought before, so start if you like to get into more details than needed. And, yeah, that webUI sucks so much....

1) You will definitely need a server to play with that
2) Then you will need robottelo to be in place: https://github.com/SatelliteQE/robottelo/
3) Then you will need to run one test in debug or whatever, just to setup all necessary things from https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/ui/test_variables.py. Just put breakpoint on for example: https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/ui/test_variables.py#L217 as test is completely non functional on 6.4, but everything you need is just all the code from https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/ui/test_variables.py#L148 to be executed
Also you uncomment teardown class for debug purposes
4) Once you stay on that breakpoint, you finally can see piece of UI that you need. Actually, it is worse than it was before, but anyway
You can go 'Configure->Classes' and choose any class you like from ui_test_variables:
![image](https://user-images.githubusercontent.com/10895065/40128474-5f61fd00-593a-11e8-88ec-3b17f9c89057.png)

There you will have first type of page that you need to automate:
![image](https://user-images.githubusercontent.com/10895065/40128538-87d5fe58-593a-11e8-82aa-5815c7f8a13f.png)

Once you create first variable, second type of page is unblocked (Configure->Smart Variables):
![image](https://user-images.githubusercontent.com/10895065/40128603-ae4300cc-593a-11e8-8b4a-6b1c0b3a432c.png)

![image](https://user-images.githubusercontent.com/10895065/40128642-ca76541a-593a-11e8-805e-77e8312a01f1.png)

We need to support both I think. Anyway start from any you like or as we don't have much choices from first

Please, refer to old implementation to get the idea: https://github.com/SatelliteQE/robottelo/blob/master/robottelo/ui/smart_variable.py
Of course, for new framework, it doesn't make to describe all the attributes and etc, but anyway just for idea

Also, of course, refer to all already implemented entities in airgun to follow the standart

**Comment by LandoCalrizzian** [comment]
@oshtaier I'll leave this issue for someone else to pick up since I have to focus on another task that will take priority. I'll try to pickup and smaller first issue to work on

**Comment by schlupov** [comment]
@oshtaier I'll work on this issue since user group entity is for someone from CF team.
